### PR TITLE
Use output mode configured in `ValueGenerator` instance

### PR DESCRIPTION
**Pull request by thomas-kl1**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | yes
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description

When I want to generate code for an array with the "single line" output, I expect the result to be on a single line. However in the case the array have deeper depth than 1, the result is on multi lines, because the original output mode is not given when generating the child nodes.

### CODE REVIEW

**Review by Ocramius — CHANGES_REQUESTED** [review]
Needs a test

**Comment by thomas-kl1** [comment]
Hi @Ocramius 
I've pushed the test as well. Do you want me to add a suppress warning for the psalm rule "unused method" on the data provider?

File: test/Generator/ValueGeneratorTest.php [hunk 1]
```diff
@@ -498,4 +497,55 @@ public function testExceptionInvalidValue(mixed $value, string $type): void
...
+    ): void {
+        $valueGenerator = new ValueGenerator($array, $type, $outputMode);
+
+        self::assertSame($valueGenerator->generate(), $output);
+    }
+
+    /**
+     * Data provider for testArrayWithOutputMode test
+     */
+    public static function multipleOutputArray(): array
```
Comment (Ocramius) [inline test/Generator/ValueGeneratorTest.php:519]: This error is expected: PHPUnit 10 is not yet understood by `psalm/plugin-phpunit`.

Re-generating the baseline should fix this.

Comment (thomas-kl1) [reply test/Generator/ValueGeneratorTest.php:519]: I'm not sure what to do with this information, can I do something there?

Comment (froschdesign) [reply test/Generator/ValueGeneratorTest.php:519]: @thomas-kl1 
See: https://psalm.dev/docs/running_psalm/dealing_with_code_issues/#using-a-baseline-file

**Review by Ocramius — APPROVED** [review]
Thanks @thomas-kl1!

**Pull request merged**